docs/guides/using-with-flakes.md

```diff
@@ -1,32 +1,46 @@
-`devenv` can be used without the `devenv` CLI by integrating into [Nix Flakes](https://www.tweag.io/blog/2020-05-25-flakes/), if you're more familiar with the Nix language/ecosystem.
+`devenv` can be integrated with [Nix Flakes](https://www.tweag.io/blog/2020-05-25-flakes/) if you're more familiar with the Nix language and ecosystem.
 
-Some usecases for using devenv configuration inside flakes is for projects that want to define other Nix flake features, apart from the development shell.
-These include a Nix package for the project, NixOS and home-manager modules related to the project.
-Usually you want to use the same lock file for the development shell as well as the Nix package and others, so that everything is based on the same nixpkgs.
+You can define your own packages, NixOS and home-manager modules, but still benefit from devenv's development features.
+The development shell will share the same inputs and lock file as all the other outputs of your flake, ensuring that your entire project is using the same `nixpkgs` revision.
 
-A Nix flake includes the inputs from `devenv.yaml` as well as the devenv configuration that you'd usually find in `devenv.nix`. `flake.lock` is the lock file for Nix flakes, the equivalent to `devenv.lock`.
+With flakes, you also no longer need dedicated configuration files for devenv:
+
+* the inputs from `devenv.yaml` are replaced by the flake's inputs
+* `devenv.nix` becomes a shell module in `flake.nix`
+* the `devenv.lock` is replaced by the `flake.lock`
 
 ## Getting started
 
-To quickly set a project up with Nix flakes, use of `nix flake init`, like:
+Set up a new project with Nix flakes using our template:
 
 ```console
 $ nix flake init --template github:cachix/devenv
 ```
 
-This will create a `flake.nix` with devenv configuration, as well as a `.envrc` direnv configuration.
+This template will create:
+
+* a `flake.nix` containing a basic devenv configuration
+* an `.envrc` to optionally set up automatic shell activation with direnv.
 
-Open the devenv shell using:
+Open the devenv shell with:
 
 ```console
-$ nix develop
+$ nix develop --impure
 ```
 
-This will create a lock file and open up a new shell that adheres to the devenv configuration stated in `flake.nix`.
+This will create a `flake.lock` lock file and open up a new shell based on the devenv configuration stated in `flake.nix`.
+
+!!! note "Why do I need `--impure`?"
+    When working with flakes, pure mode prevents devenv from accessing and modifying its state data.
+    Certain features, like running processes with `devenv up`, won't work in pure mode.
```

Comment:
> pure mode prevents devenv from accessing and modifying its state data.
It comes across as though devenv does some fiddling with state data inside Nix, which isn't the case.

Not sure at what detail `--impure` needs to be described, but I think the only use of `--impure` in devenv is that it uses the project directory path in its expressions to build configuration files. So it is able to define an environment variable `DEVENV_STATE=/path/to/the/project/.devenv/state`. Without impure it wouldn't be able to determine the absolute path of the project dir.

I don't know for sure whether we want to go into detail, so I think the current change is fine :+1: